jobserver/views/projects.py

```diff
@@ -187,9 +187,12 @@ def get_repo(repo, ctx):
                 }
 
         # use the threadpool to parallelise the repo requests
-        yield from repo_thread_pool.map(
-            get_repo, repos, itertools.repeat(ctx), timeout=30
-        )
+        try:
+            yield from repo_thread_pool.map(
+                get_repo, repos, itertools.repeat(ctx), timeout=30
+            )
+        except TimeoutError:
```

Comment:
Would this be cleaner and easier to follow if this `except` clause were part of the `try` block within `get_repo`? This localizes the try-except to the code it is protecting as much as possible. The `requests.HTTPError` and the timeout exception are similar in this context: different flavours of "the required resource was not available".

```
try:
	is_private = self.get_github_api().get_repo_is_private(
		repo.owner, repo.name
	)
except requests.HTTPError:
	is_private = None
except requests.Timeout:
	name = "GitHub API Unavailable"
	is_private = None
	url = ""
```
It's also not clear to me why the `HTTPError` block just sets `is_private = None`, whereas the new `TimeoutError` block changes the name and the URL. Could it just be one block instead as in your original commits? There may well be a good reason for that -- if so can it please be documented as it isn't obvious (to me)?

Reply:
The reason is a `TimeoutError` and not one of the exceptions from the `requests` namespace is that it's not requests nor even the GitHub API that's giving the timeout. The code before this change would have gracefully handled the 504 error (gateway timeout) previously seen as it's an `HTTPError`. The problem is/was that it can take longer than the `ThreadPoolExecutor.map()` call's configured timeout for these to appear and so the ThreadPoolExecutor throws a 'TimeoutError'. Does that make sense?

Reply:
It's not possible to localise the handling of this timeout within the get repo function that's passed to map()

Reply:
>I think it's quick and useful to add the additional exceptions to the inner except: block. (requests.ConnectionError and requests.Timeout).

I agree, I'll add that to this PR. As for the other two I think raising tickets is a great idea